File: caesar.py
from tkinter import *

# Caesar Cipher function to encrypt a message with a given shift key
def caesar_cipher_encrypt(message, key):
    encrypted_message = ""
    for char in message:
        if char.isalpha():
            char_code = ord(char)
            char_code += key
            if char.isupper():
                if char_code > ord('Z'):
                    char_code -= 26
                elif char_code < ord('A'):
                    char_code += 26
            else:
                if char_code > ord('z'):
                    char_code -= 26
                elif char_code < ord('a'):
                    char_code += 26
            encrypted_message += chr(char_code)
        else:
            encrypted_message += char
    return encrypted_message

# Caesar Cipher function to decrypt an encrypted message with a given shift key
def caesar_cipher_decrypt(encrypted_message, key):
    decrypted_message = ""
    for char in encrypted_message:
        if char.isalpha():
            char_code = ord(char)
            char_code -= key
            if char.isupper():
                if char_code < ord('A'):
                    char_code += 26
                elif char_code > ord('Z'):
                    char_code -= 26
            else:
                if char_code < ord('a'):
                    char_code += 26
                elif char_code > ord('z'):
                    char_code -= 26
            decrypted_message += chr(char_code)
        else:
            decrypted_message += char
    return decrypted_message

# GUI initialization
root = Tk()
root.title("Caesar Cipher Encryption and Decryption")
root.geometry("600x400")

# Main frame
main_frame = Frame(root, padx=20, pady=20)
main_frame.pack(expand=True)

# Title label
title_label = Label(main_frame, text="Caesar Cipher Encryption and Decryption", font=("Arial", 18, "bold"), fg='red')
title_label.pack(pady=20)

# Message input label
msg_label = Label(main_frame,fg="green", text="Enter a message:")
msg_label.pack()

# Message input field
msg_entry = Entry(main_frame, width=30)
msg_entry.pack()

# Key selection label
key_label = Label(main_frame, bg="blue", text="Select a shift key (1-9):")
key_label.pack()

# Key selection dropdown menu
key_var = StringVar(root)
key_var.set("1") # default value
key_menu = OptionMenu(main_frame, key_var, "1", "2", "3", "4", "5", "6", "7", "8", "9")
key_menu.pack()

# Encryption function
def encrypt():
    message = msg_entry.get()
    key = int(key_var.get())
    encrypted_message = caesar_cipher_encrypt(message, key)
    result_label.config(text=encrypted_message)

# Decryption function
def decrypt():
    encrypted_message = msg_entry.get()
    key = int(key_var.get())
    decrypted_message = caesar_cipher_decrypt(encrypted_message, key)
    result_label.config(text=decrypted_message)

# Button frame
button_frame = Frame(main_frame)
button_frame.pack(pady=20)

# Encrypt button
encrypt_button = Button(button_frame, text="Encrypt",bg="green", font=("Arial", 14), command=encrypt)
encrypt_button.pack(side=LEFT, padx=10)

# Decrypt button
decrypt_button = Button(button_frame, text="Decrypt",bg="red", font=("Arial", 14), command=decrypt)
decrypt_button.pack(side=LEFT, padx=10)

# Result label
result_label = Label(main_frame,width=40, height= 30,bd=3, bg="white", text="", font=("Arial", 14))
result_label.pack()


root.mainloop()

File: caesar.java
import java.awt.*;
import java.awt.event.*;
import javax.swing.*;

public class CaesarCipherGUI extends JFrame {

    // Caesar Cipher function to encrypt a message with a given shift key
    public static String caesarCipherEncrypt(String message, int key) {
        String encryptedMessage = "";
        for (char c : message.toCharArray()) {
            if (Character.isLetter(c)) {
                int charCode = (int) c;
                charCode += key;
                if (Character.isUpperCase(c)) {
                    if (charCode > (int) 'Z') {
                        charCode -= 26;
                    } else if (charCode < (int) 'A') {
                        charCode += 26;
                    }
                } else {
                    if (charCode > (int) 'z') {
                        charCode -= 26;
                    } else if (charCode < (int) 'a') {
                        charCode += 26;
                    }
                }
                encryptedMessage += (char) charCode;
            } else {
                encryptedMessage += c;
            }
        }
        return encryptedMessage;
    }

    // Caesar Cipher function to decrypt an encrypted message with a given shift key
    public static String caesarCipherDecrypt(String encryptedMessage, int key) {
        String decryptedMessage = "";
        for (char c : encryptedMessage.toCharArray()) {
            if (Character.isLetter(c)) {
                int charCode = (int) c;
                charCode -= key;
                if (Character.isUpperCase(c)) {
                    if (charCode < (int) 'A') {
                        charCode += 26;
                    } else if (charCode > (int) 'Z') {
                        charCode -= 26;
                    }
                } else {
                    if (charCode < (int) 'a') {
                        charCode += 26;
                    } else if (charCode > (int) 'z') {
                        charCode -= 26;
                    }
                }
                decryptedMessage += (char) charCode;
            } else {
                decryptedMessage += c;
            }
        }
        return decryptedMessage;
    }

    public CaesarCipherGUI() {
        setTitle("Caesar Cipher Encryption and Decryption");
        setSize(600, 400);
        setDefaultCloseOperation(JFrame.EXIT_ON_CLOSE);

        // Main panel
        JPanel mainPanel = new JPanel();
        mainPanel.setLayout(new BoxLayout(mainPanel, BoxLayout.PAGE_AXIS));
        mainPanel.setBorder(BorderFactory.createEmptyBorder(20, 20, 20, 20));
        add(mainPanel);

        // Title label
        JLabel titleLabel = new JLabel("Caesar Cipher Encryption and Decryption");
        titleLabel.setFont(new Font("Arial", Font.BOLD, 18));
        titleLabel.setForeground(Color.RED);
        titleLabel.setAlignmentX(Component.CENTER_ALIGNMENT);
        mainPanel.add(titleLabel);
        mainPanel.add(Box.createVerticalStrut(20));

        // Message input label
        JLabel msgLabel = new JLabel("Enter a message:");
        msgLabel.setForeground(Color.GREEN);
        mainPanel.add(msgLabel);
        mainPanel.add(Box.createVerticalStrut(5));

        // Message input field
        JTextField msgField = new JTextField(30);
        mainPanel.add(msgField);
        mainPanel.add(Box.createVerticalStrut(10));

        // Key selection label
        JLabel keyLabel = new JLabel("Select a shift key (1-9):");
        keyLabel.setBackground(Color.BLUE);
        keyLabel.setOpaque(true);
        mainPanel.add(keyLabel);
        mainPanel.add(Box.createVerticalStrut(5));

        // Key selection dropdown menu
            // Key selection dropdown menu
    String[] keys = {"1", "2", "3", "4", "5", "6", "7", "8", "9"};
    JComboBox<String> keyDropdown = new JComboBox<>(keys);
    mainPanel.add(keyDropdown);
    mainPanel.add(Box.createVerticalStrut(20));

    // Encrypt button
    JButton encryptButton = new JButton("Encrypt");
    encryptButton.setAlignmentX(Component.CENTER_ALIGNMENT);
    mainPanel.add(encryptButton);
    mainPanel.add(Box.createVerticalStrut(10));

    // Encrypted message label
    JLabel encryptedLabel = new JLabel("Encrypted Message:");
    encryptedLabel.setForeground(Color.BLUE);
    mainPanel.add(encryptedLabel);
    mainPanel.add(Box.createVerticalStrut(5));

    // Encrypted message display field
    JTextArea encryptedField = new JTextArea(5, 30);
    encryptedField.setEditable(false);
    JScrollPane scrollPane = new JScrollPane(encryptedField);
    mainPanel.add(scrollPane);
    mainPanel.add(Box.createVerticalStrut(20));

    // Decrypt button
    JButton decryptButton = new JButton("Decrypt");
    decryptButton.setAlignmentX(Component.CENTER_ALIGNMENT);
    mainPanel.add(decryptButton);
    mainPanel.add(Box.createVerticalStrut(10));

    // Decrypted message label
    JLabel decryptedLabel = new JLabel("Decrypted Message:");
    decryptedLabel.setForeground(Color.BLUE);
    mainPanel.add(decryptedLabel);
    mainPanel.add(Box.createVerticalStrut(5));

    // Decrypted message display field
    JTextArea decryptedField = new JTextArea(5, 30);
    decryptedField.setEditable(false);
    JScrollPane scrollPane2 = new JScrollPane(decryptedField);
    mainPanel.add(scrollPane2);

    // Action listeners
    encryptButton.addActionListener(new ActionListener() {
        public void actionPerformed(ActionEvent e) {
            String message = msgField.getText();
            int key = Integer.parseInt((String) keyDropdown.getSelectedItem());
            String encryptedMessage = caesarCipherEncrypt(message, key);
            encryptedField.setText(encryptedMessage);
        }
    });

    decryptButton.addActionListener(new ActionListener() {
        public void actionPerformed(ActionEvent e) {
            String encryptedMessage = encryptedField.getText();
            int key = Integer.parseInt((String) keyDropdown.getSelectedItem());
            String decryptedMessage = caesarCipherDecrypt(encryptedMessage, key);
            decryptedField.setText(decryptedMessage);
        }
    });

    // Display the window
    setVisible(true);
}

public static void main(String[] args) {
    new CaesarCipherGUI();
}

}